### ISSUE TITLE
`(elpy-refactor-rename)` produces "Refactoring failed for some reason"

### ISSUE THREAD

**Issue by Anarcroth** [jorgenschaefer/elpy#1880]
# Summary

First and foremost, thank you for this project! It has been a huge help for my work and I truly appreciate the amount of work that has been put into this!

Issue: when running refactoring commands, I'm getting -> "Refactoring failed for some reason", "We could not extract the selection as a function", "We could not extract the selection as a variable".

# Steps to reproduce

1. Setup elpy as per documentation (to be verified after I show my config if it's correct)
2. Setup jedi/rope/black/etc
3. Install lsp-jedi and install jedi-language-server
4. Create any python project and import it as root (as per lsp prompt)
5. Write some simple code and try to rename a local variable or a function parameter
6. Fail

# My configuration
## OS
ArchLinux 5.10.6-arch1-1
Emacs 27.1
Python 3.9.1
pip 20.2.3

## Notes
I have additionally installed [`lsp-jedi`](https://github.com/pappasam/jedi-language-server) and [`jedi-language-server`](https://github.com/pappasam/jedi-language-server).
Virtualenv is under `~/.emacs.d/envs/ENV` where I have installed `flake8` and `jedi-language-server`.
While `elpy/rpc-venv/` has the other necessary python packages (setup and installed automatically by `(elpy-config)`.

I *successfully* connect to the lsp server upon opening a python file.

## Result of `(elpy-config)`
```
Emacs.............: 27.1
Elpy..............: 1.35.0
Virtualenv........: ENV (/home/anarcroth/.emacs.d/envs/ENV/)
Interactive Python: python 3.9.1 (/home/anarcroth/.emacs.d/envs/ENV/bin/python)
RPC virtualenv....: rpc-venv (/home/anarcroth/.emacs.d/elpy/rpc-venv)
 Python...........: python 3.9.1 (/home/anarcroth/.emacs.d/elpy/rpc-venv/bin/python)
 Jedi.............: 0.18.0
 Rope.............: 0.18.0
 Autopep8.........: 1.5.4
 Yapf.............: 0.30.0
 Black............: 20.8b1
Syntax checker....: flake8 (/home/anarcroth/.emacs.d/envs/ENV/bin/flake8)
```

## Elpy configuration in my init.el (I'm adding the lsp setup as well)
```
;; Lsp
(require 'lsp-mode)
(require 'lsp-ui)
(setq lsp-enable-indentation nil)
(setq lsp-enable-completion-at-point nil)
(setq lsp-ui-peek-enable t)
(setq lsp-ui-peek-show-directory t)
(setq lsp-ui-sideline-show-hover nil)
(setq lsp-ui-sideline-show-diagnostics nil)
(setq lsp-ui-sideline-show-code-actions nil)
(setq lsp-ui-doc-enable nil)
;; Python
(add-hook 'python-mode-hook #'lsp)
(add-hook 'elpy-mode-hook #'lsp)
;; Map lsp-ui key bindings
(global-set-key (kbd "C-M-.") 'lsp-ui-peek-find-references)
(global-set-key (kbd "C-M-,") 'lsp-ui-peek-find-definitions)
(global-set-key (kbd "C-M-'") 'lsp-ui-peek-find-implementation)

;; Elpy
(require 'elpy)
(require 'lsp-jedi)
(elpy-enable)
(pyvenv-activate "~/.emacs.d/envs/ENV")
(setq elpy-rpc-python-command "python")
(setq python-check-command (expand-file-name "flake8"))
(with-eval-after-load 'flycheck
  (setq elpy-modules (delq 'elpy-module-flymake elpy-modules))
  (add-hook 'elpy-mode-hook 'flycheck-mode))
(with-eval-after-load 'lsp
  (add-to-list 'lsp-disabled-clients 'pyls)
  (add-to-list 'lsp-enabled-clients 'jedi))
```

### Demo code and Stacktrace

```
#!/usr/bin/env python

def funcname(arg): <--- put coursor on arg and do (elpy-refactor-rename) then enter new name and get stacktrace below
    print(arg)
```

```
Debugger entered--Lisp error: (error "Refactoring failed for some reason")
  signal(error ("Refactoring failed for some reason"))
  error("Refactoring failed for some reason")
  elpy-refactor-rename("arg2")
  funcall-interactively(elpy-refactor-rename "arg2")
  call-interactively(elpy-refactor-rename record nil)
  command-execute(elpy-refactor-rename record)
  counsel-M-x-action("elpy-refactor-rename")
  ivy-call()
  ivy-read("M-x " ... ... ...
```

### What I've tried

1. I looked through the issues on GitHub, but alas couldn't find the exact same problem. However, I might have missed something important....I'm not sure.
2. Downgrade to elpy 1.34.0 - result is that elpy-refactor acts in a different way, with a lot of prompts, but the actual refactor works in the end.
3. Search a lot on the internet. Try to setup jedi-language-server on different python environments - didn't work.

**Comment by sammorley-short** [comment]
I am also finding this problem, and have repro'd it with all my other configs turned off.

**Comment by falematte** [comment]
I got the same problem!

**Comment by Anarcroth** [comment]
After I tried to "debug" the issue, I found the solution -> downgrade `jedi` to `17.2`. Refactoring (`elpy-refactor-*`) (and even `elpy-goto-*`) works for me! I managed to even integrate (maybe not completely, but yet to test it) `elpy` with `lsp pylsp` without a problem.

**Fix:** `pip install jedi==0.17.2` in `rpy-env` (or your defined rpc-env).

**Related issues:** #1890 #1903 #1885 #1795

**Comments:** It looks like something is breaking the integration in `jedi=18.0`.

**Note:** I have to done extensive tests to see which feature of `elpy` work or break with `jedi=18.0`, so I cannot be a judge what part (if even any) of the integration breaks with `jedi` it seems though it's more of a common problem, since there is also an `anaconda-mode` issue [#401](https://github.com/pythonic-emacs/anaconda-mode/issues/401)

I'm closing this issue as resolved, since for me it is :)
